### PR TITLE
make router dynamic stats configurable

### PR DESCRIPTION
**Pull request by ramaraochavali**
Makes router dynamic stats configurable. 
Fixes https://github.com/istio/istio/issues/23805

### CODE REVIEW

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -314,6 +314,10 @@ var (
 	EnableServiceEntrySelectPods = env.RegisterBoolVar("PILOT_ENABLE_SERVICEENTRY_SELECT_PODS", true,
 		"If enabled, service entries with selectors will select pods from the cluster. "+
 			"It is safe to disable it if you are quite sure you don't need this feature").Get()
+
 	InjectionWebhookConfigName = env.RegisterStringVar("INJECTION_WEBHOOK_CONFIG_NAME", "istio-sidecar-injector",
 		"Name of the mutatingwebhookconfiguration to patch, if istioctl is not used.")
+
+	EnableProxyDynamicStats = env.RegisterBoolVar("PROXY_DYNAMIC_STATS", true,
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go:321]: `PILOT_ENABLE_PROXY_DYNAMIC_STATS`?

Comment (ramaraochavali) [reply pilot/pkg/features/pilot.go]: They are actually cluster stats so I changed to `PILOT_ENABLE_CLUSTER_DYNAMIC_STATS`

**Comment by howardjohn** [comment]
why not use Envoy filter?

**Comment by ramaraochavali** [comment]
I thought Envoy Filter is not possible because of this comment https://github.com/istio/istio/issues/23805#issuecomment-628389171

**Comment by howardjohn** [comment]
I was investigating this yesterday - I am fairly sure you can override
fields that are NOT set. You just cannot override a field that is
explicitly set to true to he false. But since we don't explicitly set
dynamic stats field it should be fine to override

**Comment by ramaraochavali** [comment]
I see. If it works with Envoy filter, I am OK to close this

File: pilot/pkg/networking/core/v1alpha3/filters.go [hunk 2]
```diff
@@ -50,7 +52,9 @@ var (
 	routerFilter = &http_conn.HttpFilter{
 		Name: wellknown.Router,
 		ConfigType: &http_conn.HttpFilter_TypedConfig{
-			TypedConfig: util.MessageToAny(&router.Router{}),
+			TypedConfig: util.MessageToAny(&router.Router{
+				DynamicStats: &wrappers.BoolValue{Value: features.EnableClusterDynamicStats},
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/filters.go:56]: whats the default here?

**Comment by penglei** [comment]
Yes, that's right. It's probably my fault that the original issue is based on v1.5.1 and the master branch already uses the TypedConfig field to build the router filter.
Maybe I should test it using the master branch.

File: pilot/pkg/networking/core/v1alpha3/filters.go (hunk 2 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/filters.go:56]: true -which is Envoy's default

**Comment by ramaraochavali** [comment]
@penglei No problem,. Will wait for your confirmation. If it works with Envoy filter we can probably close this

**Comment by penglei** [comment]
@ramaraochavali  I tested on the v1.6.0-rc.1 version and this problem is no longer there.  We can close this.

**Pull request closed without merging**